### PR TITLE
Image hashing take two and code coverage increase

### PR DESCRIPTION
**Pull request by asalani93**
Addresses #2110 again.

New features:
* migration actually runs on live
* can no longer upload duplicate PNGs
* 'date modified' is removed from PNGs
* summernote now works correctly when you repeatedly press save or upload a duplicate media object

I also updated selenium because it wasn't working with the new versions of Firefox.

Increased code coverage on some files as well.  Also fixed a "bug" in the fields controller as part of this.  It's a two line fix so it's OK going in here.  It's not really even a bug either, the problem was that I'd convert literally every possible set of field restrictions to strings, and then check to see if they were strings.  This allowed for really weird field restrictions that would never be encountered in normal use of the website.  The only possible way to even hit this is by thoroughly abusing the API.

### CODE REVIEW

File: app/controllers/media_objects_controller.rb [hunk 1]
```diff
@@ -101,15 +101,10 @@ def saveMedia
 
     # Rotate based on EXIF data, then strip it out.
```
Comment (bdonald25) [inline app/controllers/media_objects_controller.rb]: This comment is no longer valid.

**Comment by jaypoulz** [comment]
You're failing because of test coverage.

**Comment by asalani93** [comment]
Yes.

File: db/migrate/20150514161235_add_hash_to_media_object.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class AddHashToMediaObject < ActiveRecord::Migration
+  def isFeatured?(mo)
```
Comment (bdonald25) [inline db/migrate/20150514161235_add_hash_to_media_object.rb]: You are no longer deleting things so this function is not needed.

**Comment by bdonald25** [comment]
No need for spinning earth gif

**Pull request merged**